### PR TITLE
DAOS-9729 Test: Removed the Python queue and used the list for result checking in thread.

### PR DESCRIPTION
**Pull request by ravalsam**
Python background job is getting hung so removed the queue for result checking and instead using the list.
- Cleanup the code to use the ior local object instead of the self.ior as enospace is running multiple IOR job and it's overlapping the container UUID.
- nvme fault Test was failing as it was not using the correct container uuid. So updated code based on recent changes
- Updated nvme utility to create it's own container instead to avoid the conflicts when tests are run in threads
- Removed the skip test as DAOS-5281 as it's not valid test case.

Signed-off-by: Samir Raval <samir.raval@intel.com>

### CODE REVIEW

**Comment by phender** [comment]
Just a heads up.  When https://github.com/daos-stack/daos/pull/8521 lands you'll want to merge to pickup the new method for defining the Mpirun.

**Comment by ravalsam** [comment]
test_enospace_time_with_fgran in Ci for Five loop and it all got passed. 
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8180/47/artifact/Functional%20Hardware%20Medium/nvme/enospace.py/

**Review by rpadma2 — APPROVED** [review]

**Review by shimizukko — DISMISSED** [review]

**Review by saurabhtandan — DISMISSED** [review]
Code LGTM, but @ravalsam do you think it's worth running other nvme tests before merging this PR given there is decent changes in the utils/nvme_utils.py?

**Comment by ravalsam** [comment]
> Code LGTM, but @ravalsam do you think it's worth running other nvme tests before merging this PR given there is decent changes in the utils/nvme_utils.py?

I don't think so, as enospace.py test is using all those changes implemented in nvme_util.py.

**Comment by saurabhtandan** [comment]
> > Code LGTM, but @ravalsam do you think it's worth running other nvme tests before merging this PR given there is decent changes in the utils/nvme_utils.py?
> 
> I don't think so, as enospace.py test is using all those changes implemented in nvme_util.py.

Sure, if you are confident I am good.

**Comment by phender** [comment]
@ravalsam this PR needs to run at least the `ftest/nvme/enospace.py` test before it can land.  Since this test is a weekly test you'll need to explicitly specify the test tags in your commit message with the `Test-tag` pragma.  You should also include any tests using the methods modified in the two utils files.

**Review by phender — CHANGES_REQUESTED** [review]
@ravalsam this PR needs to run at least the `ftest/nvme/enospace.py` test before it can land - there are no `FTEST_nvme` results in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8180/50/testReport/. 

Since this test is a weekly test you'll need to explicitly specify the test tags in your commit message with the `Test-tag` pragma. You should also include any tests using the methods modified in the two utils files.

**Comment by ravalsam** [comment]
> @ravalsam this PR needs to run at least the `ftest/nvme/enospace.py` test before it can land - there are no `FTEST_nvme` results in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8180/50/testReport/.
> 
> Since this test is a weekly test you'll need to explicitly specify the test tags in your commit message with the `Test-tag` pragma. You should also include any tests using the methods modified in the two utils files.

enospace was ran part of the previous PR (Comment previously) https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8180/47/artifact/Functional%20Hardware%20Medium/nvme/enospace.py/. Those utils files are used by the enopsace.py test file so not sure if we need to run any other tests.

**Review by shimizukko — DISMISSED** [review]

**Comment by ravalsam** [comment]
All test passed except Three and new defect open for the same DAOS-10457

File: src/tests/ftest/nvme/enospace.py [hunk 1]
```diff
@@ -100,9 +100,14 @@ def ior_bg_thread(self):
...
-        self.create_cont()
-        job_manager.job.dfs_cont.update(self.container.uuid)
+
+        # create container
+        container = TestContainer(self.pool, daos_command=DaosCommand(self.bin))
+        container.get_params(self)
+        container.create()
+
         env = ior_bg_cmd.get_default_env(str(job_manager))
+        job_manager.job.dfs_cont.update(container.uuid)
```
Comment (shimizukko) [inline src/tests/ftest/nvme/enospace.py:110]:
```suggestion
        self.add_container(pool=self.pool)

        env = ior_bg_cmd.get_default_env(str(job_manager))
        job_manager.job.dfs_cont.update(self.container.uuid)
```
Can't we use add_container() (and not import `TestContainer`)?

Comment (ravalsam) [reply src/tests/ftest/nvme/enospace.py:110]: Yes if there is major review comment , I will change it but I do not want to spin again and wait for all EC NVMe tests to run which takes day. I think release/2.0 is not going to be live for that long so not sure if we want to waste time

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by ravalsam** [comment]
Three test failure in Ci are not because of the change made in PR and open new defect DAOS-10457 to track those issue.

**Comment by phender** [comment]
There is a concern that in last run release/2.0 weekly build https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing-2.0/58/testReport/FTEST_erasurecode/EcodAggregationOffRebuild/ the `erasurecode/offline_rebuild_aggregation.py:EcodAggregationOffRebuild.test_ec_offline_rebuild_agg_disabled` test passed and now it is failing in your PR.  While the `EcodAggregationOffRebuild.execution()` method does end up calling the methods modified by this PR in the src/tests/ftest/util/ec_utils.py file the failure does not appear to be influenced by the code changes.

**Comment by ravalsam** [comment]
> There is a concern that in last run release/2.0 weekly build https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing-2.0/58/testReport/FTEST_erasurecode/EcodAggregationOffRebuild/ the `erasurecode/offline_rebuild_aggregation.py:EcodAggregationOffRebuild.test_ec_offline_rebuild_agg_disabled` test passed and now it is failing in your PR. While the `EcodAggregationOffRebuild.execution()` method does end up calling the methods modified by this PR in the `src/tests/ftest/util/ec_utils.py` file the failure does not appear to be influenced by the code changes.

They both are different code base:
- This PR is based on release/2.0 commit e8ad352635baf1ee497bd9b9f1ffa9b6949d0ed3 from date Apr-26.
- The weekly-testing-2.0/58 is based on commit c4be672e3429fd77460173cc7ddd42052b3ba82f from date Apr-15.

**Comment by phender** [comment]
The following tests are affected by changes to `src/tests/ftest/util/ec_utils.py` and `src/tests/ftest/util/nvme_utils.py`.  This list includes the test results from https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-8180/62:
```
erasurecode/aggregation.py:EcodAggregationOff.test_ec_aggregation_disabled	                             PASS
erasurecode/aggregation.py:EcodAggregationOff.test_ec_aggregation_default                                    PASS
erasurecode/aggregation.py:EcodAggregationOff.test_ec_aggregation_time                                       PASS
erasurecode/multiple_failure.py:EcodOnlineMultFail.test_ec_multiple_targets_on_same_rank                     NOT A 2.0 TEST
erasurecode/multiple_failure.py:EcodOnlineMultFail.test_ec_multiple_targets_on_diff_ranks                    NOT A 2.0 TEST
erasurecode/offline_rebuild_aggregation.py:EcodAggregationOffRebuild.test_ec_offline_rebuild_agg_disabled    INTERRUPTED
erasurecode/offline_rebuild_aggregation.py:EcodAggregationOffRebuild.test_ec_offline_rebuild_agg_default     SKIP
erasurecode/offline_rebuild_aggregation.py:EcodAggregationOffRebuild.test_ec_offline_agg_during_rebuild      SKIP
erasurecode/offline_rebuild.py:EcodOfflineRebuild.test_ec_offline_rebuild                                    PASS / INTERRUPTED / INTERRUPTED
erasurecode/rebuild_disabled.py:EcodDisabledRebuild.test_ec_degrade                                          PASS / PASS / PASS
erasurecode/restart.py:EcodServerRestart.test_ec_restart_before_agg                                          PASS
erasurecode/restart.py:EcodServerRestart.test_ec_restart_after_agg                                           PASS
erasurecode/restart.py:EcodServerRestart.test_ec_restart_during_agg                                          PASS
erasurecode/hard_rebuild.py:EcodIorHardRebuild.test_ec_ior_hard_online_rebuild                               NOT A 2.0 TEST

nvme/enospace.py:NvmeEnospace.test_enospace_lazy_with_bg                                                     PASS
nvme/enospace.py:NvmeEnospace.test_enospace_lazy_with_fg                                                     PASS
nvme/enospace.py:NvmeEnospace.test_enospace_time_with_bg                                                     PASS
nvme/enospace.py:NvmeEnospace.test_enospace_time_with_fg                                                     PASS
nvme/enospace.py:NvmeEnospace.test_enospace_no_aggregation                                                   PASS
nvme/fault.py:NvmeFault.test_nvme_fault                                                                      PASS
nvme/health.py:NvmeHealth.test_monitor_for_large_pools                                                       PASS

osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain                                                  PASS
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_without_checksum                                 *NOT RUN*
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_during_aggregation                               *NOT RUN*
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_oclass                                           *NOT RUN*
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_multiple_pools                                   *NOT RUN*
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_during_rebuild                                   *NOT RUN*
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_after_snapsot                                    *NOT RUN*
osa/offline_drain.py:OSAOfflineDrain.test_osa_offline_drain_with_less_pool_space                             NOT A 2.0 TEST
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_without_checksum         PASS
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_multiple_pools           *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_server_stop              PASS
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegrate_during_rebuild             *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_oclass                   *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegrate_during_aggregation         *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegration_with_rf                  *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegrate_with_blank_node            *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegrate_after_snapshot             *NOT RUN*
osa/offline_reintegration.py:OSAOfflineReintegration.test_osa_offline_reintegrate_with_less_pool_space       NOT A 2.0 TEST
```

**Comment by phender** [comment]
@ravalsam could you please issue a run using the `Build with Parameters` option and the following in the `TestTag` entry box to verify the changes against the tests not run and a re-run of the three failing tests:
```
offline_drain offline_reintegration_full ec_offline_rebuild ec_offline_rebuild_agg_disabled
```

**Comment by ravalsam** [comment]
Three tests offline_rebuild.py and offline_rebuild_aggregation.py failed because of DAOS-10457.

Two offline_drain.py tests timeouts and not all test ran as part of weekly run so not sure what is going on https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing-2.0/58/

**Comment by ravalsam** [comment]
During second run, all the offline_drain.py test passed except test_osa_offline_drain_oclass, which is failed with No Space Error while creating pool later on in the test. It's not related to the changed made in this PR.

@daos-stack/daos-gatekeeper please check the result and merge this PR.

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Please add request_for_2.0.4 to ticket if you want to land it still. 2.0.3 window is closed and future is uncertain.

**Pull request closed without merging**